src/core/lru.h

```diff
@@ -0,0 +1,181 @@
+// Copyright 2023, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+#pragma once
+
+#include <absl/container/node_hash_map.h>
+
+#include <optional>
+
+#include "base/logging.h"
+
+namespace dfly {
+
+template <typename T> class Lru {
```

Comment:
can you please check what is the memory overhead per item for this data structure